subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/strategy/StaticVersionComparator.java

```diff
@@ -66,6 +66,9 @@ public int compare(Version version1, Version version2) {
                 return -1;
             }
             if (is1Number && is2Number) {
+                if (numericPart1 == numericPart2) {
```

Comment:
This may not be true even if the numerical values are the same since it tests for instance equality. I would recommend doing the `compareTo` and a `continue` if `0`, return the result otherwise.

Reply:
Right. My Java is a bit rusty.

Fixed now.